mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1357,6 +1397,32 @@ impl TendermintCoin {
     }
 }
 
+fn find_client(rpc_urls: Vec<String>) -> Result<(HttpClient, Vec<String>), String> {
+    let mut res_urls = rpc_urls.clone();
+    let mut errors = Vec::new();
+    let mut clients = Vec::new();
+    for (i, url) in rpc_urls.iter().enumerate() {
+        match HttpClient::new(url.as_str()) {
+            Ok(client) => clients.push(client),
+            Err(e) => {
+                errors.push(e);
+                res_urls.remove(i);
+            },
+        }
+    }
```

Comment:
I think we should return an error if we can't create an `HttpClient` from any `url`. This is used in the coin activation, and if a user provided a wrong url in the activation request, they should know this by returning an error.

Reply:
The reason I store all errors in vector is that we could have several places where error occurs during HttpClient [creation](https://github.com/KomodoPlatform/atomicDEX-API/blob/tendermint-multiple-rpcs-optimization/mm2src/coins/tendermint/rpc/tendermint_native_rpc.rs#L299), so I wanted to show it.
If its not so necessary, I can just return error like  "Invalid urls were provided", if clients vector is empty. So we don't need to keep all errors in vector and can save some resources.

Reply:
Thanks for the explanation, I agree with you, it's good to show all the errors.
What I meant is that, right now, if at least one url is valid, an `Ok` result is returned in the below line of code, but there can be other invalid urls provided in the coin activation request.
https://github.com/KomodoPlatform/atomicDEX-API/blob/af6a7e7ad7f5ea329e081a63b5df732224b50c7c/mm2src/coins/tendermint/tendermint_coin.rs#L1415
In my opinion all the urls should be valid in the coin activation request, if one or more are invalid, errors should be returned.

Reply:
Oh, sorry seems like my implementation was not working as I wanted.

I wanted to filter vector of urls from invalid ones and keep working with correct urls.
I changed a bit `find_client` function in this branch. Now its called `find_client_valid_urls` and here we remove invalid urls using `retain` function of vector. As a result `TendermintRpcClientImpl` will keep valid urls.
In addition, vector of errors was removed. I decided to just show log error with invalid url and its error. If iteration isnt successful, the error `Fail to init HttpClient...` will be returned. I think its more readable than showing all errors in one error message.

I understood you and implemented your suggestion in [this](https://github.com/KomodoPlatform/atomicDEX-API/blob/tendermint-multiple-rpcs-optimization-reject-invalid-urls/mm2src/coins/tendermint/tendermint_coin.rs#L1392) branch.

I would like to give a chance to other valid urls, if we have them, and not to return error, if we have at least one invalid url. But I would agree that this variant of implementation is a bit massive.

Please, check one more time my implementation in current branch and another from the link above. If the second variant more preferable, I will make changes in current branch.

Reply:
> I would like to give a chance to other valid urls, if we have them, and not to return error, if we have at least one invalid url.

The problem with this approach, is that if the user made a typo in one of the urls, it will be removed and not used although it should be valid if an error is returned to the user and he fixed the typo in a new request. So, I prefer to return an error to the user if there is an invalid url, it's their responsibility to either remove the invalid urls from the request or otherwise fix them.

Reply:
> So, I prefer to return an error to the user if there is an invalid url, it's their responsibility to either remove the invalid urls from the request or otherwise fix them.

I left a log for this case, but now I see, for instance users who use mobile version probably wont see logs?
Well, yeah, in this case we should return error immediately.

Reply:
Done